station/models.py

```diff
@@ -306,10 +306,15 @@ def set_permissions(self):
         if self.permissions_level == "public":
             assign_perm("view_measurements", standard_group, self)
             assign_perm("view_measurements", anonymous_user, self)
+            remove_perm("view_measurements", self.owner, self)
         elif self.permissions_level == "internal":
             assign_perm("view_measurements", standard_group, self)
+            remove_perm("view_measurements", anonymous_user, self)
+            remove_perm("view_measurements", self.owner, self)
```

Comment:
And the same here. The owner, if it exist, should always have `view` permissions regardless of the `permission_level`.

I guess it will have `view` permissions via its membership to the `standard_group`, but that's somewhat irrelevant: if the owner were to be removed from that group, for whatever reason, they should still retain full powers over the objects they owns.